src/commands/faucet.ts

```diff
@@ -30,30 +31,41 @@ const FaucetModule = (cosmosClient: CosmosClient): Command => {
       return;
     }
 
+    const discordId = interaction.user.id;
     const address = interaction.options.getString('address', true);
 
     console.debug(`Token request to address ${address}`);
+    const result = await sequelize.transaction(async (t) => {
+      const lastDrop = await Drop.findOne({
+        where: { discordId: discordId },
+        order: [['createdAt', 'DESC']],
+        lock: true,
+        skipLocked: true,
+        transaction: t,
+      });
+      if (lastDrop
+        && (lastDrop.get('createdAt') as Date) > new Date(Date.now() - Config.faucet.cooldownInDay * 24 * 60 * 60 * 1000)
+      ) {
+        const txHash = lastDrop.get('txHash');
+        return `:negative_squared_cross_mark: Each user can only claim once, your previous txhash: [${txHash}](${Config.faucet.restUrl}/cosmos/tx/v1beta1/txs/${txHash}).`;
+      }
 
-    const rateLimited = rateLimiter.get(address);
-    if (rateLimited) {
-      await interaction.editReply(
-        `:negative_squared_cross_mark: Token already sent to address: \`${address}\`, Please come back again in a day`,
-      );
-      return;
-    }
-
-    try {
-      const txHash = await cosmosClient.distribute(address);
-      rateLimiter.limit(address);
-      await interaction.editReply(
-        `:white_check_mark: Transaction submitted, txhash: [${txHash}](${Config.faucet.restUrl}/cosmos/tx/v1beta1/txs/${txHash})`,
-      );
-    } catch (err: unknown) {
-      console.error(`Failed to create transaction to ${address} = `, err);
-      await interaction.editReply(
-        `:warning: Failed to create transaction to \`${address}\`, Please try again later`,
-      );
-    }
+      try {
+        const txHash = await cosmosClient.distribute(address);
+        await Drop.create({
```

Comment:
i think i prefer to create the drop with placeholder tx hash first and commit and tx. Sending the tx might prolong the lock unnecessarily